walk-in-interview/src/main/webapp/new-job/script.js

```diff
@@ -229,42 +325,69 @@ function validateRequiredUserInput() {
   const expiry = document.getElementById('expiry').valueAsNumber;
 
   if (name.value === '') {
-      setErrorMessage(/* errorMessageElementId= */'error-message', /* msg= */ name.placeholder);
-      return false;
+    setErrorMessageAndField(/* msg */ name.placeholder,
+        /** includes default msg */ true, /** error element id */ 'title');
+    return false;
   }
 
   if (description.value === '') {
-      setErrorMessage(/* errorMessageElementId= */'error-message', /* msg= */ description.placeholder);
-      return false;
+    setErrorMessageAndField(/* msg */ description.placeholder,
+        /** includes default msg */ true,
+        /** error element id */ 'description');
+    return false;
   }
 
   if (address.value === '') {
-      setErrorMessage(/* errorMessageElementId= */'error-message', /* msg= */ address.placeholder);
-      return false;
+    setErrorMessageAndField(/* msg */ address.placeholder,
+        /** includes default msg */ true, /** error element id */ 'address');
+    return false;
+  }
+
+  if (postalCode.value === '' ||
+    (parseInt(postalCode.value[0]) > JAVA_INTEGER_MAX_VALUE) ||
```

Comment:
it's not obvious why postal code first 2 characters have to be digits. you should document that. Also you are copying the parsing of postal code with 0 and 82. I would recommend instead you try to parse postalCode and if the parsing fails then setErrorMessageAndField instead of re-validating the values here.